var/spack/repos/builtin/packages/nvshmem/package.py

```diff
@@ -38,12 +40,23 @@ class Nvshmem(MakefilePackage, CudaPackage):
         when="@2.6:",
         description="Build with support for GPU initiated communication",
     )
+    variant(
+        "ibrc",
+        default=False,
+        when="@2.6:",
+        description="Build with support for IBRC remote transport",
+    )
     conflicts("~cuda")
 
+    build_system("makefile", "cmake", default="makefile")
+
     def url_for_version(self, version):
         ver_str = "{0}".format(version)
         directory = ver_str.split("-")[0]
-        url_fmt = "https://developer.download.nvidia.com/compute/redist/nvshmem/{0}/source/nvshmem_src_{1}.txz"
+        if self.spec.satisfies("@2.9:"):
+            url_fmt = "https://developer.download.nvidia.com/compute/redist/nvshmem/{0}/source/nvshmem_src_{1}.tar.xz"
```

Comment:
Sorry about this regression. I can work internally to fix this filename for 2.9 and going forward.